Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/clip.h

```diff
@@ -433,6 +575,144 @@ bool clip(      TriangleMesh& tm,
+  * @param np_s optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamBegin{vertex_point_map}
+  *     the property map with the points associated to the vertices of `tm` (`splitter`).
+  *     If this parameter is omitted, an internal property map for
+  *     `CGAL::vertex_point_t` must be available in `TriangleMesh`
+  *   \cgalParamEnd
+  *   \cgalParamBegin{visitor} a class model of `PMPCorefinementVisitor`
+  *                            that is used to track the creation of new faces.
+  *   \cgalParamEnd
+  *   \cgalParamBegin{throw_on_self_intersection} if `true`,
+  *      the set of triangles closed to the intersection of `tm` and `splitter` will be
+  *      checked for self-intersections and `CGAL::Polygon_mesh_processing::Corefinement::Self_intersection_exception`
+  *      will be thrown if at least one is found.
+  *   \cgalParamEnd
+  * \cgalNamedParamsEnd
+*/
+
+template <class TriangleMesh,
+          class NamedParameters1,
+          class NamedParameters2>
+void split(      TriangleMesh& tm,
+                 TriangleMesh& splitter,
+           const NamedParameters1& np_tm,
+           const NamedParameters2& np_s)
+{
+
+  using parameters::get_parameter;
+  using parameters::choose_parameter;
+  namespace PMP = CGAL::Polygon_mesh_processing;
+
+  typedef typename GetVertexPointMap<TriangleMesh, NamedParameters1>::type VPM1;
+  typedef typename GetVertexPointMap<TriangleMesh, NamedParameters2>::type VPM2;
+
+  typedef typename boost::template property_map<TriangleMesh, CGAL::dynamic_edge_property_t<bool> >::type Ecm;
+
+  VPM1 vpm_tm = choose_parameter(get_parameter(np_tm, internal_np::vertex_point),
+                                 get_property_map(vertex_point, tm));
+  VPM2 vpm_s = choose_parameter(get_parameter(np_s, internal_np::vertex_point),
+                                get_property_map(vertex_point, splitter));
+
+  Ecm ecm  = get(CGAL::dynamic_edge_property_t<bool>(), tm);
+
+  // create a constrained edge map and corefine input mesh with the splitter,
+  // and mark edges
+
+  PMP::corefine(tm, splitter,
+                CGAL::parameters::vertex_point_map(vpm_tm).edge_is_constrained_map(ecm),
+                CGAL::parameters::vertex_point_map(vpm_s));
+
+  //split mesh along marked edges
+  internal::split_along_edges(tm, ecm, vpm_tm);
+}
+
+
+
+/**
+  * \ingroup PMP_corefinement_grp
+  * adds intersection edges of `plane` and `tm` in `tm` and duplicates those edges.
+  *
+  * \note In the current implementation it is not possible to set the vertex point map and the default will be used.
+  * \pre \link CGAL::Polygon_mesh_processing::does_self_intersect() `!CGAL::Polygon_mesh_processing::does_self_intersect(tm)` \endlink
+  *
+  * @tparam TriangleMesh a model of `MutableFaceGraph`, `HalfedgeListGraph` and `FaceListGraph`
+  *                      An internal property map for `CGAL::vertex_point_t` must be available.
```

Comment:
Should be in the currently missing {vertex_point_map} section of NPs

Reply:
Not really, it is important that the type has an internal map because it will be used to convert the plane into a splitter, but the user cannot have an existing map for an internal not yet existing mesh (the plane).